Epix-Incorporated/Adonis
# [Big bugfix&improvement pack]: Fixes many annoying bugs

Fixes:
- Fix broken atmosphere handling
- Fix atmosphere for client
- Fix TextChat command suggestions not respecting permission level - Fixes #1600
- Fix poison command only working once - Fixes #1607
- Fix private info not showing for self in `!profile` - Fixes #1578
- Fix Dex disappearing after respawn - Fixes #1608
- Fix Dex sometimes not fetching newest APIs and Dex plugin delaying Adonis boot
- Fix friends page old online API not working anymore - Fixes https://devforum.roblox.com/t/updating-friends-api-deprecating-isonline-property-from-getfriendsasync/3171177
- Fix `!fps` updating infrequently - Fixes #1555
- Fix other misc bug fixes and optimizations with audioplayer and overhaul some parts of it. Commit for easier reviewing due to file replacement: https://github.com/Epix-Incorporated/Adonis/commit/e37abf91a372b4e24d0d6a8be7f45dba79bc8352
	- Fix audio player not skipping to next track - Fixes #1574
	- Fix audio player forward and backwards buttons not working
	- Fix audio player position slider being broken
	- Fix audio player Adonis audio adding game audios
	- Fix audio player volume controls being way too sensitive
	- Fix misc bugs in audioplayer and optimize it
- Fix `:zawarudo` audios being broken - Fixes: [![Zawarudo has 2 broken audios](https://github.com/user-attachments/assets/fba6f8a4-64e4-4c17-8284-48f8a6225d2a)](https://discord.com/channels/81902207070380032/1238053069367939133/1238053069367939133)
- Fix `:clearguis` not removing Dex
- Fix some themes Hints and Notifs not respecting `TopBarShift` and having a weird position
- Fix console not respecting TopBar size
- Fix `:showlogs` not working. Someone changed the argument order a few years ago, this changes it back. - Fixes: [![showlogs command doesn't work](https://github.com/user-attachments/assets/d2a57366-a707-4323-a422-894933922ea0)](https://discord.com/channels/81902207070380032/1214141268045729842/1214141268045729842)
- Fix notification icon missing for Aero theme
- Fix notification icon missing for Windows XP theme
- Fix `:sing` not working sometimes in game and fix some rojo and model issues. - Fixes #1664 Fixes: [![Fix sing command](https://github.com/user-attachments/assets/76593276-104f-4535-b427-92f3e2a71287)](https://discord.com/channels/81902207070380032/1209965618229674034/1209965618229674034)
- Fix `:backupmap` not backup up gravity or lighting
- Fix invalid/weird brush position for `!paint`
- Fix Unity theme `functionify` lint error
- Fix nightlymode not being used in nightly branch builds
- Fix backup loader not working
- Fix `!serverdetails` showing incorrect physics FPS
- Fix `!serverdetails` not working in studio
- Fix non-player aliases not working
- Fix not all outputs respecting `Settings.SilentStartup`
- Fix render resolution showing decimals for pixel amount in `!profile`
- Fix `service.GetProductInfo` being broken due to invalid cache timestamp
- Fix invalid string pattern for numbered player selector
- Fix `:starterscripts` invalid `table.insert` arguments
- Fix `:giveppoints` not sending a notification. Roblox has fully bricked the API now.
- Misc fixes

Changes:
- Add `:UnZaWarudo`
- Make `!paint` use `EditableImage`s. This adds a speed boost for large amounts of pixels
- Allow users to upload their own creations in `!paint` as Roblox decals
- Redo `AudioLib.luau` from the ground up to make it work
- Change Music Player dependencies `Signal` to Adonis GoodSignal and `Slider` to newest version. Fixes some breaking bugs
- Add loading animation to indicate audios loading in music player
- Add a much better version system
- Make unknown user logs a lot more user friendly.
- Convert many things to use `Settings.Message`s
- Move `:testfilter` to the debug plugin
- `:debugtesterror` from `Creators` to `Players`
- `:debugtestbiglist` from `Creators` to `Admins`
- `:debuganti` from `Creators` to `Head Admins`
- Optimise `:jail` color arg parsing
- Optimise `Functions.ArgsToString`
- Optimise `Admin.FormatCommandArguments`, `Admin.FormatCommand` and `Admin.FormatCommandAdminLevel`, improves Adonis speed considerably
- Optimise `service.FormatNumber`
- Optimise `UserPanel.luau` tab to string
- Convert `AudioPlayer.rbxmx` to raw files instead of XML
- Convert `:size` to use a variable for default max size

Removed:
- Remove `service.GetUserType`. It's a useless pre-FilteringEnabled relic
- Remove `client.TestEvent`. It's useless
- Remove `ChatCreateRobloxCommands` from `Settings` as it's no longer needed due to the bug being fixed.
- Remove `service.OwnsAsset`, just an unused alias for `service.CheckAssetOwnership`
- Remove usage of `Anti.ClassName` from `UI.lua`. It's a deprecated and nonexistent pre-FilteringEnabled relic
- Remove `service.IsLocked`, another pre-FilteringEnabled relic
- Remove `:flatten` unused `CharacterMesh` check
- Remove unused `doPcall`
- Remove unused `enumMats` in `:material`
- Remove `Functions.CatchError`. Totally useless and dublicated by `Pcall`
- Delete `Topbar.rbxmx` from `Unity`
- Delete severely out of date Audio Player copy from `Unity`
- Remove legacy references of TopBars
- Misc bloat&cruft

Minor misc changes:
- Fix streaming enabled and archivable for `:jail`
- Fix streaming enabled and archivable for `:clone`/`:loadavatar`
- Fix streaming enabled and archivable for `:ufo`
- Fix streaming enabled for `:bot`
- Fix streaming enabled for `:nuke`
- Fix `:pipe` archivable
- Fix AudioLib archivable
- Fix `:sfling` archivable
- Fix `:clown` archivable
- Fix TextChatCommands archivable
- Fix `:tornado` name
- Fix `:chik3n` parent
- Convert some `math.random()` to use `HttpService:GenerateGUID()`

> I said let him cook
